tests/check_options.py

```diff
@@ -1,149 +1,259 @@
+    print no_upload_output
+    tar_regex = re.compile(r".*tar\.gz.*")
+    if re.search(tar_regex, no_upload_output) is not None and re.search("Upload completed successully!", no_upload_output) is None:
+        tar_file = no_upload_output.split("See Insights data in ")[1]
+        print tar_file
+    else:
+        print err
+        pytest.fail("The upload was successfull despite of the option --no-upload")
+
+    os.system("rm -rf  /var/tmp/%s" % tar_file.split("/")[3])
+    print "Deleted tar file %s" %tar_file
 
 def test_no_tar_file_option():
-	no_tar_file = subprocess.Popen(["redhat-access-insights --no-tar-file"], stdout=subprocess.PIPE, shell=True)
-	no_tar_output, err = no_tar_file.communicate()
-	print no_tar_output
-	tar_regex = re.compile(r".*tar\.gz.*")
-	if re.search(tar_regex, no_tar_output) is None and re.search("Upload completed successully!", no_tar_output) is None:
-		no_tar_dir = no_tar_output.split("See Insights data in ")[1]
-		print no_tar_dir
-	else:
-		print err
-		pytest.fail("The tar file should not have been created with usage of option --no-tar-file")
-
-	os.system("rm -rf  /var/tmp/%s" % no_tar_dir.split("/")[3])
-	print "Deleted tar file %s" %no_tar_dir
+    no_tar_file = subprocess.Popen([constants.app_name + " --no-tar-file"], stdout=subprocess.PIPE, shell=True)
+    no_tar_output, err = no_tar_file.communicate()
+    print no_tar_output
+    tar_regex = re.compile(r".*tar\.gz.*")
+    if re.search(tar_regex, no_tar_output) is None and re.search("Upload completed successully!", no_tar_output) is None:
+        no_tar_dir = no_tar_output.split("See Insights data in ")[1]
+        print no_tar_dir
+    else:
+        print err
+        pytest.fail("The tar file should not have been created with usage of option --no-tar-file")
+
+    os.system("rm -rf  /var/tmp/%s" % no_tar_dir.split("/")[3])
+    print "Deleted tar file %s" %no_tar_dir
+
+def test_support_option():
+    if os.path.isfile(constants.default_log_file):
+        os.system("mv " + constants.default_log_file + " "  + constants.default_log_file + ".backup1")
+    subprocess.call([constants.app_name + " --support"], shell=True)
+    if os.path.isfile(constants.default_log_file):
+        os.system("rm -f  " + constants.default_log_file)
+        os.system("mv " + constants.default_log_file + ".backup1 "  + constants.default_log_file)
+        print "\n'" + constants.default_log_file + "' was created"
+    else:
+        pytest.fail("'" + constants.default_log_file + "' was not created")
+
+def test_version_option():
+    version = subprocess.Popen([constants.app_name + " --version"], stdout=subprocess.PIPE, shell=True)
+    version_output, err = version.communicate()
+    print version_output
+    version_regex = re.compile(r"[0-9]\.[0-9]\.[0-9]")
+    if re.search(version_regex, version_output) is not None:
+        pass
+    else:
+        print err
+        pytest.fail("Did not print version number")
+
+def test_update_collection_rules():
```

Comment:
I would probably use some sort of creation & modification times for this for the rules files but I guess this works.